lib/files.ncl

```diff
@@ -72,7 +83,12 @@ let regenerate_files | Files -> nix.derivation.Derivation
           }
             file_descr.materialisation_method
         in
-        nix-s%"regenerate_function "%{copy_command}" "%{file_descr.file}" "%{file_descr.target}""%
+        let shell_escape = fun path =>
+          path
+          |> std.string.replace "\\" "\\\\"
+          |> std.string.replace m%"""% m%"\""%
+        in
+        nix-s%"regenerate_function "%{copy_command}" "%{file_descr.file}" "%{shell_escape file_descr.target}""%
```

Comment:
I wonder if we shouldn't use `'` instead of `"` to also avoid unexpected shell expansions (what if the file name is like `$(/path/to/reverse/sell)` ?). Then we would have to escape `'` instead of `"`.

```suggestion
        let shell_escape = fun path =>
          path
          |> std.string.replace "\\" "\\\\"
          |> std.string.replace "'" "\\'"
        in
        nix-s%"regenerate_function "%{copy_command}" '%{file_descr.file}' '%{shell_escape file_descr.target}'"%
```

Reply:
I intend to use `$'…'` (dollar-sign with *single*-quotes), instead of `"…"` (*double*-quotes), to quote the escaped string. This handles `$` in the string to escape correctly.